crates/ruma-client-api/src/config/set_global_account_data.rs

```diff
@@ -49,7 +51,25 @@ pub mod v3 {
 
     impl<'a> Request<'a> {
         /// Creates a new `Request` with the given data, event type and user ID.
-        pub fn new(data: &'a RawJsonValue, event_type: &'a str, user_id: &'a UserId) -> Self {
+        ///
+        /// # Errors
+        ///
+        /// Since `Request` stores the request body in serialized form, this function can fail if
+        /// `T`s [`Serialize`][serde::Serialize] implementation can fail.
+        pub fn new<T: GlobalAccountDataEventContent>(
+            data: &'a T,
+            event_type: &'a str,
```

Comment:
This should not be a separate parameter, the event type should be taken from `data`.

Reply:
Ah of course.